examples/apps/color_finder.py

```diff
@@ -0,0 +1,740 @@
+from cozmo.lights import Color, Light
+try:
+    from PIL import Image, ImageColor, ImageDraw
+except ImportError:
+    sys.exit('Cannot import from PIL: Do `pip3 install --user Pillow` to install')
+    
+# map_color_to_light (dict): maps each color name with its cozmo.lights.Light value.
+# Red, green, and blue lights are already defined as constants in lights.py, 
+# but we need to define our own custom Light for yellow.
+
+map_color_to_light = {
+'green' : cozmo.lights.green_light, 
+'yellow' : Light(Color(name='yellow', rgb = (255, 255, 0))), 
+'blue' : cozmo.lights.blue_light, 
+'red' : cozmo.lights.red_light
+}
+
+# color_dict (dict): map of color names to regions in color space.
+# Instead of defining a color as a single (R, G, B) point, 
+# colors are defined with a minimum and maximum value for R, G, and B.
+# For example, a point with (R, G, B) = (40, 15, 225) falls entirely in the 'blue' region, 
+# because 0 < 40 < 50, 0 < 15 < 50, and 210 < 225 < 255.
+# A point with (R, G, B) = (95, 240, 160) does not fall exactly in one color region.
+# But applying color_distance_sqr between this color and all the colors in color_dict
+# will show that (95, 240, 160) is closest to the 'green' region.
+
+color_dict = {
+'red' : (210, 255, 0, 50, 0, 50), 
+'green' : (0, 50, 210, 255, 0, 50), 
+'blue' : (0, 50, 0, 50, 210, 255), 
+'white' : (200, 255, 200, 255, 200, 255), 
+'black' : (0, 30, 0, 30, 0, 30),
+'yellow' : (210, 255, 210, 255, 0, 50), 
+}
+
+# color_distance_sqr (fn (color, color_range) => int): returns the squared 
+# euclidean distance of color to color_range.
+
+def color_distance_sqr(color, color_range):
+    r, g, b = color
+    minR, maxR, minG, maxG, minB, maxB = color_range
+    rdist_sq = 0
+    gdist_sq = 0
+    bdist_sq = 0
+    if r < minR:
+        rdist_sq = (minR-r)**2
+    if r > maxR:
+        rdist_sq = (maxR-r)**2
+    if g < minG:
+        gdist_sq = (minG-g)**2
+    if g > maxG:
+        gdist_sq = (maxG-g)**2
+    if b < minB:
+        bdist_sq = (minB-b)**2
+    if b > maxB:
+        bdist_sq = (maxB-b)**2
+    return rdist_sq+gdist_sq+bdist_sq
+
+# color_balance (fn PIL image => PIL image): balances the color in an image.
+# Adjusts the color data in the image_array representation of an image so that
+# the average R, G, B values across the entire image end up equal.
+# This is called a 'gray-world' algorithm, because the colors
+# with equal R, G, B values fall along the grayscale.
+
+def color_balance(image):
+    image_array = from_pil(image)
+    image_array = image_array.transpose(2, 0, 1).astype(numpy.uint32)
+    average_g = numpy.average(image_array[1])
+    image_array[0] = numpy.minimum(image_array[0] * (average_g / numpy.average(image_array[0])), 255)
+    image_array[2] = numpy.minimum(image_array[2] * (average_g / numpy.average(image_array[2])), 255)
+    return to_pil(image_array.transpose(1, 2, 0).astype(numpy.uint8))
+
+# from_pil (fn PIL image => image array): converts PIL image to image array.
+
```

Comment:
This comment should instead be a docstring inside the method like:
'''Converts PIL image to image array.'''